### PR TITLE
triton/apps/mpi: Adding a new page for MPI installations

### PR DESCRIPTION
**Pull request by simo-tuomisto**
This PR does the following:

- Adds a new page that tells about available MPI installations.
- Updates the parallel tutorial with new information.
- Updates multiple module names in examples.

### CODE REVIEW

File: triton/apps/mpi.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+===
+
+Message Passing Interface (MPI) is used in high-performance computing (HPC)
+clusters to facilitate big parallel jobs that utilize multiple compute nodes.
+
+MPI and Slurm
+*************
+
+For a tutorial on how to do Slurm reservations for MPI jobs, check out the
+:ref:`MPI section<parallel-mpi>` of the parallel computing-tutorial.
```
Comment (rkdarst) [inline triton/apps/mpi.rst:12]:
```suggestion
:ref:`MPI section <parallel-mpi>` of the parallel computing-tutorial.
```

File: triton/apps/mpi.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+For example, one could use an
+:doc:`Intel compiler </triton/apps/intelcompilers>`
+to compile the Hello world!-example by setting ``OMPI_MPICC``- and
+``OMPI_MPIFC``-environment variables.
+
+.. tabs::
+
+  .. group-tab:: C
+
+    Intel C compiler is `icc`:
```
Comment (rkdarst) [inline triton/apps/mpi.rst:77]:
```suggestion
    Intel C compiler is ``icc``:
```

File: triton/apps/mpi.rst [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      module load openmpi/4.0.5
+      module load intel-oneapi-compilers/2021.4.0
+
+      export OMPI_MPICC=icc  # Overwrite the C compiler
+
+      mpicc    -O2 -g hello_mpi.c -o hello_mpi
+
+  .. group-tab:: Fortran
+
+    Intel Fortran compiler is `ifort`:
```
Comment (rkdarst) [inline triton/apps/mpi.rst:91]:
```suggestion
    Intel Fortran compiler is ``ifort``:
```

File: triton/examples/mpi/hello.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  module load gcc/8.4.0      # GCC
+  module load openmpi/4.0.5  # OpenMPI
+
+Compiling the code:
+
+.. tabs::
+
+  .. group-tab:: C
+
+    C code is compiled with `mpicc`:
```
Comment (rkdarst) [inline triton/examples/mpi/hello.rst:18]:
```suggestion
    C code is compiled with ``mpicc``:
```

File: triton/examples/mpi/hello.rst [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    .. code-block:: bash
+
+      cd hpc-examples/hello_mpi/
+      mpicc    -O2 -g hello_mpi.c -o hello_mpi
+
+  .. group-tab:: Fortran
+
+    Fortran code is compiled with `mpifort`:
```
Comment (rkdarst) [inline triton/examples/mpi/hello.rst:27]:
```suggestion
    Fortran code is compiled with ``mpifort``:
```

**Review by rkdarst — APPROVED** [review]
there probably is no time for this, but I would be interested in an introduction section that explains *why* MPI libraries are a thing and why it needs so much attention.  Things like how it is closely connected to the system itself, so you often need to compile specifically for it.

File: triton/examples/mpi/hello.rst [hunk 6]
```diff
@@ -0,0 +1,57 @@
+The following example uses example codes stored in
+`the hpc-examples-repository <https://github.com/AaltoSciComp/hpc-examples>`_.
+You can get the repository with the following command::
+
+  git clone https://github.com/AaltoSciComp/hpc-examples/
+
+Loading module::
+
+  module load gcc/8.4.0      # GCC
+  module load openmpi/4.0.5  # OpenMPI
```
Comment (tpfau) [inline triton/examples/mpi/hello.rst:10]: Shouldn't the OpenMPI, if it is kind of linked to a compiler anyways, automatically load the corresponding compiler? Just wondering.

**Comment by simo-tuomisto** [comment]
Double apostrophe problems should be fixed by now.

File: triton/tut/parallel.rst [hunk 7]
```diff
@@ -214,122 +213,91 @@ you should load the same compiler that you used for compiling the code.
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-  srun --time=00:05:00 --mem-per-cpu=200M --ntasks=4 ./hello_mpi
+In many cases you might require more than one node during your job's runtime.
 
-Running an MPI code in the batch mode:
+When this is the case, it is usually recommended to split the number of
+workers evenly among the nodes. To do this, one can use
+``-N N``/``--nodes=N`` and ``--ntasks-per-node=n``. For example, you could
+distribute the previously requested four tasks to two nodes with:
```
Comment (tpfau) [inline triton/tut/parallel.rst:245]: What's the actual advantage of this? i.e. balancing on different nodes. Your CPU power will stay the same (or be different depending on the node), and the only direct advantage I can see is not to be killed as quick if you run out of your memory quota...

**Review by tpfau — COMMENTED** [review]
Overall, looks good. 
I'm just wondering why distributing jobs is actually beneficial

**Pull request merged**